File: agu_chat/android/app/src/main/kotlin/com/example/agu_chat/model/Message.kt
package com.example.agu_chat.model

data class Message(
    val id: Int,
    val senderId: String,
    val senderUsername: String,
    val senderPhotoUrl: String,
    val senderDisplayName: String,
    val recipientUsername: String,
    val recipientDisplayName: String,
    val recipientPhotoUrl: String,
    val content: String,
    val messageSent: String
) {

    val isIncoming: Boolean
        get() = senderUsername != Constanst.currentUsername
}

File: ReactClient/src/models/message.ts
export interface IMessage{
    id: string;
    senderId: string;
    senderUsername: string;
    senderPhotoUrl: string;
    senderDisplayName: string;
    recipientId: string;
    recipientUsername: string;
    recipientPhotoUrl: string;
    recipientDisplayName: string;
    content: string;
    dateRead?: Date;
    messageSent: Date;
}